### PR TITLE
feat(plugin): CustomBadges

### PR DESCRIPTION
**Pull request by PFearr**
Just a simple plugin that allows the user to create/remove custom badges to the user of their choice.

https://github.com/Vendicated/Vencord/assets/66143801/0217a36a-2c24-435a-b1d5-ca571412ae42

### CODE REVIEW

**Comment by ryanccn** [comment]
https://github.com/Vendicated/Vencord/pull/375#issuecomment-1586122860

> Badges are a way to denote people who contributed in certain ways

**Comment by ActuallyTheSun** [comment]
> https://github.com/Vendicated/Vencord/pull/375#issuecomment-1586122860
> 
> > Badges are a way to denote people who contributed in certain ways

that's unrelated. since these badges are local only, they don't need some special significance, they're just there for you.

that plugin got denied because it added badges for *all* users

File: src/plugins/customBadges.tsx [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+
+    stop() {
+        removeContextMenuPatch("user-context", this.userContextMenuPatch);
+        Object.entries(indicatorLocations).forEach(([_, value]) => {
+            value.onDisable();
+        });
+    },
+
+
+    options: {
```
Comment (ActuallyTheSun) [inline src/plugins/customBadges.tsx:391]: You should use `definePluginSettings` to add settings to a plugin (you can see how it works from other plugins). This allows for a top level `settings` variable which makes your code way cleaner

File: src/plugins/customBadges.tsx [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+    authors: [{ name: "PFearr", id: 147436995679879168n }],
+    dependencies: ["MessageDecorationsAPI", "MemberListDecoratorsAPI"],
+
+    userContextMenuPatch: makeContextMenuPatch("user-profile", MenuItemParentType.User),
+
+    start() {
+        const settings = Settings.plugins.CustomBadge;
+        const { displayMode } = settings;
+
+        // transfer settings from the old ones, which had a select menu instead of booleans
```
Comment (ActuallyTheSun) [inline src/plugins/customBadges.tsx:368]: ctrl+c ctrl+v moment

File: src/plugins/customBadges.tsx [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+            badgeInfo={validBadges[num]}
+        />
+    ));
+
+    if (!Object.keys(validBadges).length || Object.keys(validBadges).length === 0) return null;
+
+    return (
+        <span
+            className="customBadge"
+            style={{
```
Comment (ActuallyTheSun) [inline src/plugins/customBadges.tsx:104]: please use a `style.css` file for this

File: src/plugins/customBadges.tsx [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+        else if (childId === "roles" && props.guildId)
+            // "roles" may not be present due to the member not having any roles. In that case, add it above "Copy ID"
+            children.splice(-1, 0, <Menu.MenuGroup>{item}</Menu.MenuGroup>);
+    };
+}
+
+export default definePlugin({
+    name: "CustomBadges",
+    description: "Create custom badges for your friends or anyone!",
+    authors: [{ name: "PFearr", id: 147436995679879168n }],
```
Comment (ryanccn) [inline src/plugins/customBadges.tsx:359]: add yourself to Devs

**Review by ryanccn — CHANGES_REQUESTED** [review]

File: src/plugins/customBadges.tsx [hunk 5]
```diff
@@ -0,0 +1,399 @@
...
+        Object.entries(indicatorLocations).forEach(([_, value]) => {
+            value.onDisable();
+        });
+    },
+
+
+    options: {
+        messages: {
+            type: OptionType.BOOLEAN,
+            description: "Next the the username in messages",
```
Comment (ryanccn) [inline src/plugins/customBadges.tsx:394]: This doesn't look right

**Comment by SomeAspy** [comment]
> > is there a reason why this was denied
> 
> Badges are a way to denote people who contributed in certain ways, whether that be via contributing, donating or possible more ways
> 
> Adding badges from all mods combined is counter-intuitive, slightly confusing and also fairly pointless

https://github.com/Vendicated/Vencord/pull/375#issuecomment-1586122860

duplicate of pr#375

**Comment by Osinted** [comment]
Is this still working?

**Comment by Vendicated** [comment]
hi, thank you for your contribution!

i don't think this is that useful, if anything it should be part of a bigger EditUsers plugin. you also haven't addressed any of the feedback

thus i'll be closing this, sorry

**Pull request closed without merging**